Feature/extensions

wenzhixin opened the pull request:
* Fixed CSS error
* Fixed cookie search cannot work bug
* Fixed toolbar display error

fix #4035

TODO:

- [x] bootstrap-table-reorder-columns cannot work.

djhvscf commented on src/extensions/cookie/bootstrap-table-cookie.js:
```diff
@@ -329,7 +329,7 @@ $.BootstrapTable = class extends $.BootstrapTable {
     super.onSearch(event)
 
-    if ($(event.currentTarget).parent().hasClass('search')) {
+    if (this.options.search) {
```
Is this change equivalent?

wenzhixin replied:
Fix `$(event.currentTarget).parent().hasClass('search')` always return `false` bug.

djhvscf commented on src/extensions/sticky-header/bootstrap-table-sticky-header.css:
```diff
@@ -17,6 +17,6 @@
 }
 
-.fix-sticky table thead.thead-light {
+.fix-sticky table thead.thead-dark {
```
Why is this change? The hex color is the same

wenzhixin replied:
The last version was wrong: https://github.com/wenzhixin/bootstrap-table/blob/develop/src/extensions/sticky-header/bootstrap-table-sticky-header.css#L16

UtechtDustin commented:
I will review it as soon as WIP is removed.

UtechtDustin commented:
> Fixed cookie search cannot work bug

This Bug #4158 ?

UtechtDustin commented:
> bootstrap-table-reorder-columns cannot work.

Fixed, demo: https://live.bootstrap-table.com/code/UtechtDustin/35

UtechtDustin commented:
fix #4035

wenzhixin commented:
Fixed export dropdown error.
error: https://live.bootstrap-table.com/code/wenzhixin/44
fixed: https://live.bootstrap-table.com/code/wenzhixin/45 

![image](https://user-images.githubusercontent.com/2117018/59893854-edb59a00-9410-11e9-889a-44062ca4b23f.png)

djhvscf commented:
> > bootstrap-table-reorder-columns cannot work.
> 
> Fixed, demo: https://live.bootstrap-table.com/code/UtechtDustin/35

This is fixed. Please test it under the same live example above

UtechtDustin commented on src/extensions/reorder-columns/bootstrap-table-reorder-columns.js:
```diff
@@ -135,16 +135,19 @@ BootstrapTable.prototype.makeRowsReorderable = function () {
 
-      that.columns = that.columns.concat(columns)
+      for (let i = 0; i < ths.length; i++ ) {
+        that.fieldsColumnsIndex[ths[i]] = i
```
Can be done in the for loop at line 138, no need for an new for loop.
